src/cli/pm_pkg_command.zig

```diff
@@ -0,0 +1,784 @@
+            if (exists) {
+                return try removeProperty(allocator, root, current_key);
+            }
+            return false;
+        }
+
+        const nested_obj = root.get(current_key);
+        if (nested_obj == null or nested_obj.?.data != .e_object) {
+            return false;
+        }
+
+        var nested = nested_obj.?;
+        const deleted = try deleteNested(allocator, &nested, remaining_path);
+
+        if (deleted) {
+            try root.data.e_object.put(allocator, current_key, nested);
+        }
+
+        return deleted;
+    }
+
+    fn removeProperty(allocator: std.mem.Allocator, obj: *js_ast.Expr, key: []const u8) !bool {
+        if (obj.data != .e_object) return false;
+
+        const old_props = obj.data.e_object.properties.slice();
+        var found = false;
+        for (old_props) |prop| {
+            if (prop.key) |k| {
+                switch (k.data) {
+                    .e_string => |s| {
+                        if (strings.eql(s.data, key)) {
+                            found = true;
+                            break;
+                        }
+                    },
+                    else => {},
+                }
+            }
+        }
+
+        if (!found) return false;
+        var new_props = try allocator.alloc(js_ast.G.Property, old_props.len - 1);
+        var new_index: usize = 0;
+
+        for (old_props) |prop| {
+            if (prop.key) |k| {
+                switch (k.data) {
+                    .e_string => |s| {
+                        if (strings.eql(s.data, key)) {
+                            continue;
+                        }
+                    },
+                    else => {},
+                }
+            }
+            new_props[new_index] = prop;
+            new_index += 1;
+        }
+        const new_list = js_ast.G.Property.List.init(new_props);
+        obj.data.e_object.properties = new_list;
+
+        return true;
+    }
+
+    fn savePackageJson(allocator: std.mem.Allocator, path: []const u8, root: js_ast.Expr, pkg: *const PackageJson) !void {
+        const preserve_newline = pkg.contents.len > 0 and pkg.contents[pkg.contents.len - 1] == '\n';
+
+        var buffer_writer = JSPrinter.BufferWriter.init(allocator);
+        try buffer_writer.buffer.list.ensureTotalCapacity(allocator, pkg.contents.len + 1);
+        buffer_writer.append_newline = preserve_newline;
+
+        var writer = JSPrinter.BufferPrinter.init(buffer_writer);
+
+        _ = JSPrinter.printJSON(
+            @TypeOf(&writer),
+            &writer,
+            root,
+            &pkg.source,
+            .{
+                .indent = pkg.indentation,
+                .mangled_props = null,
+            },
+        ) catch |err| {
+            Output.errGeneric("Failed to serialize package.json: {s}", .{@errorName(err)});
+            Global.exit(1);
+        };
+
+        const content = writer.ctx.writtenWithoutTrailingZero();
+        std.fs.cwd().writeFile(.{
```

Comment:
can we use bun.sys.File.writeFile here?

Reply:
I remember I tried to use something like that for my `bun pm version` pr, but ended up getting windows build failures with u16 and u32 differences so went back to this api that did work.